src/main/scala/io/iohk/ethereum/blockchain/data/GenesisDataLoader.scala

```diff
@@ -134,15 +118,33 @@ class GenesisDataLoader(
         Failure(new RuntimeException("Genesis data present in the database does not match genesis block from file." +
           " Use different directory for running private blockchains."))
       case None =>
-        // using empty namespace because ephemDataSource.storage already has the namespace-prefixed keys
-        ephemDataSource.storage.toSeq.grouped(dbConfig.batchSize).foreach(toStore => dataSource.update(IndexedSeq(), Nil, toStore))
+        ephemDataSource.getAll(nodeStorage.namespace)
+          .foreach { case (key, value) => blockchain.saveNode(ByteString(key.toArray[Byte]), value.toArray[Byte], 0) }
```

Comment:
as far as I remember that grouping was added to (significantly?) improve performance. Don't we need it anymore?